soh/src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c

```diff
@@ -744,6 +760,11 @@ void DoorWarp1_AdultWarpIdle(DoorWarp1* this, PlayState* play) {
     Audio_PlayActorSound2(&this->actor, NA_SE_EV_WARP_HOLE - SFX_FLAG);
 
     if (DoorWarp1_PlayerInRange(this, play)) {
+        // Pause timer and set boss timestamp as soon as player steps into the blue warp.
+        if (gSaveContext.isBossRush && play->sceneNum != SCENE_KENJYANOMA) {
+            BossRush_HandleCompleteBoss(play);
+        }
+
```

Comment:
Wouldn't this make it so that this function gets called twice since it also seems to get called at the end of every boss fight? Or am I misreading that?